lib/uplinkc/object.go

```diff
@@ -230,14 +229,22 @@ func download(bucketRef C.BucketRef, path *C.char, cErr **C.char) (downloader C.
 
 	scope := bucket.scope.child()
 
-	rc, err := bucket.NewReader(scope.ctx, C.GoString(path))
```

Comment:
Still use `bucket.NewReader`, there would be a different method `bucket.NewRangeReader`. Currently `DownloadRange` has special arguments and goes through ranger rather than using stream directly.

Reply:
the point of this pr is to remove bucket.NewReader